src/nnet3/nnet-utils.cc

```diff
@@ -101,9 +215,20 @@ static void ComputeSimpleNnetContextForShift(
     input.indexes.push_back(Index(n, t));
     output.indexes.push_back(Index(n, t));
   }
-  // the assumption here is that the network just requires the ivector at time
-  // t=0.
-  ivector.indexes.push_back(Index(n, 0));
+  // push the indexes for ivector(s)
+  int32 t_modulus = GetTimePeriodForIvector(nnet);
```

Comment:
once again this code should change as the single/multiple ivector determination has nothing to do with the t_modulus value.

Also call it ivector_period or ivector_interval, same as everywhere else. Calling it t_modulus makes no sense to the reader as we didn't define it anywhere.